syndeo-template/README.md

```diff
@@ -18,46 +18,95 @@ transforms. Then, the template will take the pipeline spec, and
+mvn -B spotless:apply compile -f pom.xml -pl syndeo-template/pom.xml
 ```
 
-Write out the pipeline spec for testing: This command will generate a configuration proto for a pipeline that reads
-from a Pubsub topic, and writes avro files into a GCS bucket.
+### Run tests (unit and integration tests)
+
+Note that syndeo depends of the teleport integration testing framework, so make sure to install that locally:
 
-```shell
-mvn compile exec:java -Dexec.mainClass="com.google.cloud.syndeo.WritePipelineSpecForTesting"
 ```
+mvn install -DskipTests -f pom.xml -pl .,it/pom.xml
+```
+
+To **run unit tests** for the Syndeo template, run the following command. Note that this command knows to skip
+integration tests and only runs unit tests:
 
-Generate the template:
 ```shell
-# Generate the JAR file under target/syndeo-template-1.0-SNAPSHOT.jar
-mvn package -DskipTests
+mvn clean package test -f pom.xml -pl syndeo-template/pom.xml
 ```
 
-Run the template:
+To set up your Google Cloud project for the integration tests, the following steps assume you have installed and setup [gcloud](https://cloud.google.com/sdk/gcloud).
+
+1. Set your default project.
+
 ```shell
-# Running locally
-mvn exec:java -Dexec.mainClass="com.google.cloud.syndeo.SyndeoTemplate" \
-  -Dexec.args="--runner=DataflowRunner --project=$GCP_PROJECT --region=$REGION \
-               --gcpTempLocation=$GCS_BUCKET --pipelineSpec=$PATH_TO_SPEC --tempLocation=$GCS_BUCKET/temp"
+gcloud config set project PROJECT
+```
+
+#### BigQuery to BigTable tests
+
+The file `BigTableWriteIT` **holds integration tests** for the basic BigTable Syndeo integration. These integration tests
+rely on the existence of a BigTable instance and table, as well as a BigQuery dataset, which holds the BQ data that
+is part of the pipeline's read.
+
+This integration test requires a BigQuery dataset and a BigTable instance, as well as GCS buckets to handle artifacts.
+
+1. Create the BigQuery dataset.
+
+```
+bq mk syndeo_dataset
+```
+
+2. Create the BigTable instance.
+
+```
+gcloud bigtable instances create teleport --display-name=teleport --cluster-config=id=teleport,zone=us-central1-a
+```
+
+3. Create the artifact bucket
 
-# Running a pushed template (see Push template below)
-Run Template on Dataflow from Uploaded Template:
-gcloud dataflow flex-template run "syndeojob-`date +%Y%m%d-%H%M%S`" \
-  --template-file-gcs-location "gs://$GCS_BUCKET/template.json" --region $REGION \
-  --temp-location "gs://$GCS_BUCKET/temp" --parameters=pipelineSpec="gs://$GCS_BUCKET/pubsub_to_avro_config.txt"
 ```
+ARTIFACT_BUCKET=[CHANGE ME]
+gsutil mb gs://$ARTIFACT_BUCKET
+```
+
+4. Create the temporary location bucket
+
+```
+TEMP_LOCATION_BUCKET=[CHANGE ME]
+gsutil mb gs://$TEMP_LOCATION_BUCKET
+```
+
+5. Run the end-to-end test.
+
+The below command tests the workflow by publishing some data to BQ and checking that it makes it over to Bigtable.
+
+
+```shell
+mvn clean package test -f pom.xml -pl syndeo-template/pom.xml  \
```

Comment:
Everything in this README worked up until this point.  The fork/branch was cloned/checked out in the Google Cloud Shell environment.

```
git log --oneline
290b65a1 (HEAD -> bq-2-bt, origin/bq-2-bt) found right series of incantations
```

This command resulted in:
<details>
<summary>java.lang.ExceptionInInitializerError</summary>

```
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running com.google.cloud.syndeo.it.BigTableWriteIT
Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 3.558 sec <<< FAILURE!
testBigQueryToBigTableSmallNonTemplateJob(com.google.cloud.syndeo.it.BigTableWriteIT)  Time elapsed: 3.478 sec  <<< ERROR!
java.lang.ExceptionInInitializerError
        at com.google.api.services.bigquery.Bigquery$Builder.build(Bigquery.java:5251)
        at org.apache.beam.sdk.io.gcp.bigquery.BigQueryServicesImpl$DatasetServiceImpl.<init>(BigQueryServicesImpl.java:594)
        at org.apache.beam.sdk.io.gcp.bigquery.BigQueryServicesImpl$DatasetServiceImpl.<init>(BigQueryServicesImpl.java:537)
        at org.apache.beam.sdk.io.gcp.bigquery.BigQueryServicesImpl.getDatasetService(BigQueryServicesImpl.java:190)
        at org.apache.beam.sdk.io.gcp.bigquery.BigQueryIO$Write.validate(BigQueryIO.java:2650)
        at org.apache.beam.sdk.transforms.PTransform.validate(PTransform.java:174)
        at org.apache.beam.sdk.Pipeline$ValidateVisitor.enterCompositeTransform(Pipeline.java:662)
        at org.apache.beam.sdk.runners.TransformHierarchy$Node.visit(TransformHierarchy.java:581)
        at org.apache.beam.sdk.runners.TransformHierarchy$Node.visit(TransformHierarchy.java:585)
        at org.apache.beam.sdk.runners.TransformHierarchy$Node.access$500(TransformHierarchy.java:240)
        at org.apache.beam.sdk.runners.TransformHierarchy.visit(TransformHierarchy.java:214)
        at org.apache.beam.sdk.Pipeline.traverseTopologically(Pipeline.java:469)
        at org.apache.beam.sdk.Pipeline.validate(Pipeline.java:598)
        at org.apache.beam.sdk.Pipeline.run(Pipeline.java:322)
        at org.apache.beam.sdk.Pipeline.run(Pipeline.java:309)
        at com.google.cloud.syndeo.transforms.BigTableSchemaTransformTest.setUpBigQueryResources(BigTableSchemaTransformTest.java:128)
        at com.google.cloud.syndeo.it.BigTableWriteIT.testBigQueryToBigTableSmallNonTemplateJob(BigTableWriteIT.java:58)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
        at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
        at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
        at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
        at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
        at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
        at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
        at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
        at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
        at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:242)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:137)
        at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
        at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
        at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
Caused by: java.lang.IllegalStateException: You are currently running with version 2.0.0 of google-api-client. You need at least version 1.15 of google-api-client to run version 1.25.0 of the BigQuery API library.
        at com.google.common.base.Preconditions.checkState(Preconditions.java:534)
        at com.google.api.client.util.Preconditions.checkState(Preconditions.java:113)
        at com.google.api.services.bigquery.Bigquery.<clinit>(Bigquery.java:48)
        ... 50 more


Results :

Tests in error:
  testBigQueryToBigTableSmallNonTemplateJob(com.google.cloud.syndeo.it.BigTableWriteIT)

Tests run: 1, Failures: 0, Errors: 1, Skipped: 0

```
</details>

Reply:
this issue did not repro for me. Maybe we can leave this one as is? wdyt?